File: app/src/main/java/com/hosigus/askposts/ui/activity/AnswerDetailActivity.java
package com.hosigus.askposts.ui.activity;

import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.support.constraint.ConstraintLayout;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MotionEvent;
import android.view.View;
import android.view.WindowManager;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.hosigus.askposts.App;
import com.hosigus.askposts.R;
import com.hosigus.askposts.config.NetUrls;
import com.hosigus.askposts.item.Option.BaseOption;
import com.hosigus.askposts.item.jsonBean.BaseBean;
import com.hosigus.askposts.item.jsonBean.QuestionDetail;
import com.hosigus.askposts.ui.view.CircleImageView;
import com.hosigus.imageloader.ImageLoader;
import com.hosigus.simplerecycleadapter.BaseAdapter;
import com.hosigus.simplerecycleadapter.BaseHolder;
import com.hosigus.tools.interfaces.NetCallback;
import com.hosigus.tools.items.JSONBean;
import com.hosigus.tools.utils.NetUtils;
import com.hosigus.tools.utils.ToastUtils;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class AnswerDetailActivity extends AppCompatActivity {

    public static void start(Context context, String answerJson,QuestionDetail detail) {
        Intent intent = new Intent(context, AnswerDetailActivity.class);
        intent.putExtra("answerJson", answerJson);
        intent.putExtra("kind", detail.getKind());
        intent.putExtra("title", detail.getTitle());
        context.startActivity(intent);
    }

    private String kind,title;
    private boolean isChangingLike,isCommending;
    private QuestionDetail.AnswersBean answer;
    private List<QuestionDetail.AnswersBean.Commend> commendList;
    private BaseAdapter<QuestionDetail.AnswersBean.Commend> adapter;

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_share_report, menu);
        return true;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_answer_detail);

        Toolbar bar = findViewById(R.id.toolbar);
        bar.setTitle("");
        setSupportActionBar(bar);
        bar.setNavigationOnClickListener(v -> finish());

        initData();
    }

    private void initData() {
        try {
            kind = getIntent().getStringExtra("kind");
            title = getIntent().getStringExtra("title");
            answer = new QuestionDetail.AnswersBean(new JSONObject(getIntent().getStringExtra("answerJson")));
        } catch (JSONException e) {
            e.printStackTrace();
            ToastUtils.show("出错");
            finish();
            return;
        }
        NetUtils.post(new BaseOption(NetUrls.GET_COMMEND_LIST)
                .addUserParam()
                .param("answer_id", answer.getId()), new NetCallback() {
            @Override
            public void onSuccess(JSONBean data) {
                BaseBean baseBean = (BaseBean) data;
                if (baseBean.isGoodJson()) {
                    commendList = new ArrayList<>();
                    try {
                        JSONArray commendArray = new JSONArray(baseBean.getData());
                        for (int i = 0; i < commendArray.length(); i++) {
                            commendList.add(new QuestionDetail.AnswersBean.Commend(commendArray.getJSONObject(i)));
                        }
                        initView();
                    } catch (JSONException e) {
                        e.printStackTrace();
                        ToastUtils.show("服务器错误");
                        finish();
                    }
                } else {
                    ToastUtils.show("服务器错误");
                    finish();
                }
            }

            @Override
            public void onFail(Exception e) {
                ToastUtils.show("网络错误");
                finish();
            }
        });
    }

    private void initView() {
        TextView likeV, commendV,sendV;

        ConstraintLayout commendCL = findViewById(R.id.cl_commend);
        View mask = findViewById(R.id.mask);
        EditText commend = findViewById(R.id.et_commend);
        likeV = findViewById(R.id.tv_left);
        commendV = findViewById(R.id.tv_right);
        sendV = findViewById(R.id.btn_send);

        Drawable light = getResources().getDrawable(R.drawable.ic_good_light);
        Drawable dark = getResources().getDrawable(R.drawable.ic_good_dark2);
        light.setBounds(0, 0, light.getMinimumWidth(), light.getMinimumHeight());
        dark.setBounds(0, 0, dark.getMinimumWidth(), dark.getMinimumHeight());

        if (answer.isPraised()) {
            likeV.setCompoundDrawables(light, null, null, null);
        }

        likeV.setOnClickListener(v->{
            if (isChangingLike) {
                return;
            }
            isChangingLike = true;
            if (!answer.isPraised()) {
                NetUtils.post(new BaseOption(NetUrls.LIKE)
                        .addUserParam()
                        .param("answer_id", answer.getId()), new NetCallback() {
                    @Override
                    public void onSuccess(JSONBean data_like) {
                        isChangingLike = false;
                        if (((BaseBean) data_like).isStatusOK()) {
                            likeV.setCompoundDrawables(light, null, null, null);
                            answer.setIs_praised(1);
                            answer.setPraise_num(answer.getPraise_num() + 1);
                            likeV.setText(String.valueOf(answer.getPraise_num()));
                        } else {
                            ToastUtils.show("点赞失败");
                        }
                    }

                    @Override
                    public void onFail(Exception e) {
                        isChangingLike = false;
                        ToastUtils.show("服务器错误");
                    }
                });
            } else {
                NetUtils.post(new BaseOption(NetUrls.UNLIKE)
                        .addUserParam()
                        .param("answer_id", answer.getId()), new NetCallback() {
                    @Override
                    public void onSuccess(JSONBean data_like) {
                        isChangingLike = false;
                        if (((BaseBean) data_like).isStatusOK()) {
                            likeV.setCompoundDrawables(dark, null, null, null);
                            answer.setIs_praised(0);
                            answer.setPraise_num(answer.getPraise_num() - 1);
                            likeV.setText(String.valueOf(answer.getPraise_num()));
                        } else {
                            ToastUtils.show("取消失败");
                        }
                    }

                    @Override
                    public void onFail(Exception e) {
                        isChangingLike = false;
                        ToastUtils.show("服务器错误");
                    }
                });
            }
        });

        mask.setOnClickListener(v -> {
            commend.clearFocus();
        });
        commend.setOnFocusChangeListener((v, hasFocus) -> {
            InputMethodManager imm = (InputMethodManager) AnswerDetailActivity.this.getSystemService(Context.INPUT_METHOD_SERVICE);
            if (!hasFocus&&imm!=null) {
                imm.hideSoftInputFromWindow(commendCL.getWindowToken(), 0);
                commendCL.setVisibility(View.GONE);
                mask.setVisibility(View.GONE);
            }
        });
        sendV.setOnClickListener(v -> {
            if (isCommending) {
                return;
            }
            String commendStr = commend.getText().toString();
            if (commendStr.isEmpty()) {
                ToastUtils.show("评论为空");
            }
            isCommending = true;
            NetUtils.post(new BaseOption(NetUrls.COMMEND_ANSWER)
                    .addUserParam()
                    .param("answer_id", answer.getId())
                    .param("content", commendStr),
                    new NetCallback() {
                        @Override
                        public void onSuccess(JSONBean data) {
                            isCommending = false;
                            if (((BaseBean) data).isStatusOK()) {
                                commend.setText("");
                                commend.clearFocus();
                                initData();
                            } else {
                                ToastUtils.show("评论失败");
                            }
                        }

                        @Override
                        public void onFail(Exception e) {
                            isCommending = false;
                            ToastUtils.show("网络错误");
                        }
            });
        });
        commendV.setOnClickListener(v -> {
            InputMethodManager imm = (InputMethodManager) AnswerDetailActivity.this.getSystemService(Context.INPUT_METHOD_SERVICE);
            if (imm != null) {
                commendCL.setVisibility(View.VISIBLE);
                mask.setVisibility(View.VISIBLE);
                commend.setFocusable(true);
                commend.requestFocus();
                imm.showSoftInput(commend,0);
            }
        });

        commendList.add(0, null);
        adapter = new BaseAdapter<QuestionDetail.AnswersBean.Commend>(commendList) {
            private static final int TYPE_HEAD = 1;

            @Override
            protected int getResId(int viewType) {
                switch (viewType) {
                    case TYPE_HEAD:
                        return R.layout.item_answer_detail_head;
                    default:
                        return R.layout.item_commend;
                }
            }

            @Override
            public int getItemViewType(int position) {
                return position == 0 ? TYPE_HEAD : 0;
            }

            @Override
            protected BaseHolder<QuestionDetail.AnswersBean.Commend> onCreateHolder(View itemView) {
                return new BaseHolder<QuestionDetail.AnswersBean.Commend>(itemView) {
                    @Override
                    public void bindData(QuestionDetail.AnswersBean.Commend data) {
                        switch (getItemViewType()) {
                            case TYPE_HEAD: {
                                if (kind.equals("情感")) {
                                    View sex = getView(R.id.v_sex);
                                    if (answer.getGender().equals("男")) {
                                        sex.setVisibility(View.VISIBLE);
                                        sex.setBackgroundResource(R.drawable.ic_sex_boy);
                                    } else if (answer.getGender().equals("女")) {
                                        sex.setVisibility(View.VISIBLE);
                                        sex.setBackgroundResource(R.drawable.ic_sex_girl);
                                    }
                                }
                                if (!answer.getPhoto_thumbnail_src().isEmpty()) {
                                    CircleImageView head = getView(R.id.civ_head);
                                    ImageLoader.with(AnswerDetailActivity.this)
                                            .place(R.drawable.ic_default_head)
                                            .error(R.drawable.ic_default_head)
                                            .load(answer.getPhoto_thumbnail_src().replace("http://", "https://"))
                                            .into(head);
                                }
                                TextView name = getView(R.id.tv_name),
                                        time = getView(R.id.tv_time),
                                        topic = getView(R.id.tv_topic),
                                        title = getView(R.id.tv_title),
                                        describe = getView(R.id.tv_describe);
                                title.setText(AnswerDetailActivity.this.title);
                                describe.setText(AnswerDetailActivity.this.answer.getContent());
                                String topicStr = "#" + kind + "#";
                                topic.setText(topicStr);
                                time.setText(answer.getCreated_at().split(" ")[0]);
                                name.setText(answer.getNickname());

                                List<String> photoList = answer.getPhoto_url();
                                List<ImageView> photoVList = new ArrayList<>();
                                switch (photoList.size()) {
                                    case 4:
                                        ImageView p4 = getView(R.id.iv_4);
                                        photoVList.add(p4);
                                    case 3:
                                        ImageView p3 = getView(R.id.iv_3);
                                        photoVList.add(p3);
                                    case 2:
                                        ImageView p2 = getView(R.id.iv_2);
                                        photoVList.add(p2);
                                    case 1:
                                        ImageView p1 = getView(R.id.iv_1);
                                        photoVList.add(p1);
                                }
                                for (int i = 0; i < photoList.size(); i++) {
                                    photoVList.get(i).setVisibility(View.VISIBLE);
                                    ImageLoader.with(AnswerDetailActivity.this).load(photoList.get(i).replace("http://", "https://")).into(photoVList.get(i));
                                }
                                TextView num = getView(R.id.tv_answer_num);
                                String numStr = "评论" + String.valueOf(commendList.size() - 1);
                                num.setText(numStr);
                                if (commendList.size() - 1 != 0) {
                                    RelativeLayout rl = getView(R.id.rl_no_answer);
                                    rl.setVisibility(View.GONE);
                                }
                                break;
                            }
                            case 0: {
                                if (kind.equals("情感")) {
                                    View sex = getView(R.id.v_sex);
                                    if (data.getGender().equals("男")) {
                                        sex.setVisibility(View.VISIBLE);
                                        sex.setBackgroundResource(R.drawable.ic_sex_boy);
                                    } else if (data.getGender().equals("女")) {
                                        sex.setVisibility(View.VISIBLE);
                                        sex.setBackgroundResource(R.drawable.ic_sex_girl);
                                    }
                                }
                                if (!data.getPhoto_thumbnail_src().isEmpty()) {
                                    CircleImageView head = getView(R.id.civ_head);
                                    ImageLoader.with(AnswerDetailActivity.this)
                                            .place(R.drawable.ic_default_head)
                                            .error(R.drawable.ic_default_head)
                                            .load(data.getPhoto_thumbnail_src().replace("http://", "https://"))
                                            .into(head);
                                }
                                TextView name = getView(R.id.tv_name),
                                        describe = getView(R.id.tv_describe),
                                        time = getView(R.id.tv_time);
                                name.setText(data.getNickname());
                                describe.setText(data.getContent());
                                time.setText(data.getCreated_at().split(" ")[0]);
                            }
                        }
                    }
                };
            }
        };
        RecyclerView rv = findViewById(R.id.rv_question_detail);
        rv.setLayoutManager(new LinearLayoutManager(this));
        rv.setAdapter(adapter);
    }
}

File: app/src/main/java/com/hosigus/askposts/ui/activity/AnswerDetailActivity.kt
package com.hosigus.askposts.ui.activity

import android.app.Dialog
import android.app.ProgressDialog
import android.content.Context
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.view.Menu
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import com.hosigus.askposts.R
import com.hosigus.askposts.config.COMMEND_ANSWER
import com.hosigus.askposts.config.GET_COMMEND_LIST
import com.hosigus.askposts.config.LIKE
import com.hosigus.askposts.config.UNLIKE
import com.hosigus.askposts.item.Option.BaseNetOption
import com.hosigus.askposts.item.jsonBean.BaseBean
import com.hosigus.askposts.item.jsonBean.QuestionDetail
import com.hosigus.askposts.utils.setHeadImage
import com.hosigus.askposts.utils.setSex
import com.hosigus.imageloader.ImageLoader
import com.hosigus.simplerecycleadapter.BaseAdapter
import com.hosigus.simplerecycleadapter.BaseHolder
import com.hosigus.tools.interfaces.NetCallback
import com.hosigus.tools.items.JSONBean
import com.hosigus.tools.utils.NetUtils
import com.hosigus.tools.utils.ToastUtils
import kotlinx.android.synthetic.main.activity_answer_detail.*
import org.json.JSONArray
import org.json.JSONObject

/**
 * Created by 某只机智 on 2018/6/4.
 */
class AnswerDetailActivity : AppCompatActivity() {
    private lateinit var kind:String
    private lateinit var title:String

    private lateinit var answer:QuestionDetail.AnswersBean
    private var isChangingLike: Boolean = false
    private var adapter: BaseAdapter<QuestionDetail.AnswersBean.Commend> = object :BaseAdapter<QuestionDetail.AnswersBean.Commend>(){
        private val TYPE_HEAD = 0
        private val TYPE_DEFAULT = 100
        override fun getItemViewType(position: Int): Int {
            return when (position) {
                0 -> TYPE_HEAD
                else -> TYPE_DEFAULT
            }
        }

        override fun getResId(viewType: Int): Int {
            return when (viewType) {
                TYPE_DEFAULT -> R.layout.item_commend
                TYPE_HEAD -> R.layout.item_answer_detail_head
                else -> -1
            }
        }
        override fun onCreateHolder(itemView: View?): BaseHolder<QuestionDetail.AnswersBean.Commend> {
            return object : BaseHolder<QuestionDetail.AnswersBean.Commend>(itemView) {
                override fun bindData(data: QuestionDetail.AnswersBean.Commend) {
                    when (itemViewType) {
                        TYPE_DEFAULT -> {
                            setSex(getView(R.id.v_sex), data.gender!!, kind)
                            setHeadImage(this@AnswerDetailActivity,
                                    getView(R.id.civ_head), data.photo_thumbnail_src!!)
                            val name: TextView = getView(R.id.tv_name)
                            val describe: TextView = getView(R.id.tv_describe)
                            val time: TextView = getView(R.id.tv_time)
                            name.text = data.nickname
                            describe.text = data.content
                            time.text = data.created_at!!.split(" ")[0]
                        }
                        TYPE_HEAD -> {

                            setSex(getView(R.id.v_sex), answer.gender!!, kind)
                            setHeadImage(this@AnswerDetailActivity,
                                    getView(R.id.civ_head), answer.photo_thumbnail_src!!)

                            val name: TextView = getView(R.id.tv_name)
                            val time: TextView = getView(R.id.tv_time)
                            val topic: TextView = getView(R.id.tv_topic)
                            val title: TextView = getView(R.id.tv_title)
                            val describe: TextView = getView(R.id.tv_describe)

                            name.text = answer.nickname
                            time.text = answer.created_at!!.split(" ")[0]
                            val topicStr = "#$kind$#"
                            topic.text = topicStr
                            title.text = this@AnswerDetailActivity.title
                            describe.text = answer.content

                            val photoUrlList: ArrayList<String> = ArrayList()
                            val photoVList: ArrayList<ImageView> = ArrayList()

                            if (photoUrlList.size > 0) {
                                val p1 = getView<ImageView>(R.id.iv_1)
                                photoVList.add(p1)
                            }
                            if (photoUrlList.size > 1) {
                                val p2 = getView<ImageView>(R.id.iv_2)
                                photoVList.add(p2)
                            }
                            if (photoUrlList.size > 2) {
                                val p3 = getView<ImageView>(R.id.iv_3)
                                photoVList.add(p3)
                            }
                            if (photoUrlList.size > 3) {
                                val p4 = getView<ImageView>(R.id.iv_4)
                                photoVList.add(p4)
                            }

                            for (i in photoUrlList.indices) {
                                photoVList[i].visibility = View.VISIBLE
                                ImageLoader.with(this@AnswerDetailActivity).load(photoUrlList.get(i).replace("http://", "https://")).into(photoVList[i])
                            }
                            val num = getView<TextView>(R.id.tv_answer_num)
                            val numStr = "评论" + (itemCount - 1).toString()
                            num.text = numStr
                            if (itemCount != 1) {
                                val rl = getView<RelativeLayout>(R.id.rl_no_answer)
                                rl.visibility = View.GONE
                            }
                        }
                    }
                }
            }
        }
    }

    private lateinit var dialog:Dialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_answer_detail)

        toolbar.title = ""
        setSupportActionBar(toolbar)
        toolbar.setNavigationOnClickListener { finish() }

        initData()
        initView()
        refreshCommendData()

    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_share_report, menu)
        return true
    }

    private fun initData() {
        try {
            kind = intent.getStringExtra("kind")
            title = intent.getStringExtra("title")
            answer = QuestionDetail.AnswersBean(JSONObject(intent.getStringExtra("answerJson")))
        } catch (e:Exception) {
            ToastUtils.show("出错")
            finish()
        }
    }

    private fun initView() {
        val changeLikeBtn: TextView = findViewById(R.id.tv_left)
        val commendBtn: TextView = findViewById(R.id.tv_right)

        dialog = ProgressDialog.show(this, "正在提交评论", "请稍候")
        dialog.dismiss()

        val light: Drawable = ContextCompat.getDrawable(this, R.drawable.ic_good_light)
        val dark: Drawable = ContextCompat.getDrawable(this, R.drawable.ic_good_dark2)
        light.setBounds(0, 0, light.minimumWidth, light.minimumHeight)
        dark.setBounds(0, 0, dark.minimumWidth, dark.minimumHeight)

        if (answer.isPraised()) {
            changeLikeBtn.setCompoundDrawables(light, null, null, null)
        }

        changeLikeBtn.setOnClickListener{
            if (isChangingLike) {
                return@setOnClickListener
            }
            isChangingLike = true
            if (!answer.isPraised()) {
                NetUtils.post(
                        BaseNetOption(LIKE)
                                .addUserParam()
                                .param("answer_id", answer.id),
                        object : NetCallback {
                            override fun onSuccess(data: JSONBean) {
                                isChangingLike = false
                                if ((data as BaseBean).isStatusOK()) {
                                    changeLikeBtn.setCompoundDrawables(light, null, null, null)
                                    answer.is_praised = 1
                                    answer.praise_num += 1
                                    changeLikeBtn.text = answer.praise_num.toString()
                                } else {
                                    ToastUtils.show("点赞失败")
                                }
                            }

                            override fun onFail(e: java.lang.Exception) {
                                isChangingLike = false
                                ToastUtils.show("服务器错误")
                            }
                        }
                )
            } else {
                NetUtils.post(
                        BaseNetOption(UNLIKE)
                                .addUserParam()
                                .param("answer_id", answer.id),
                        object : NetCallback {
                            override fun onSuccess(data: JSONBean?) {
                                isChangingLike = false
                                if ((data as BaseBean).isStatusOK()) {
                                    changeLikeBtn.setCompoundDrawables(dark, null, null, null)
                                    answer.is_praised = 0
                                    answer.praise_num -= 1
                                    changeLikeBtn.text = answer.praise_num.toString()
                                } else {
                                    ToastUtils.show("取消失败")
                                }
                            }

                            override fun onFail(e: java.lang.Exception?) {
                                isChangingLike = false
                                ToastUtils.show("服务器错误")
                            }

                        }
                )
            }
        }

        mask.setOnClickListener { et_commend.clearFocus() }
        et_commend.setOnFocusChangeListener { _, hasFocus ->
            val imm: InputMethodManager = this.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            if (!hasFocus) {
                imm.hideSoftInputFromWindow(cl_commend.windowToken, 0)
                cl_commend.visibility = View.GONE
                mask.visibility = View.GONE
            }
        }
        commendBtn.setOnClickListener {
            val imm: InputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            cl_commend.visibility = View.VISIBLE
            mask.visibility = View.VISIBLE
            et_commend.isFocusable = true
            et_commend.requestFocus()
            imm.showSoftInput(et_commend, 0)
        }
        btn_send.setOnClickListener {
            val commendStr = et_commend.text.toString()
            if (commendStr.isBlank()) {
                ToastUtils.show("评论为空")
                return@setOnClickListener
            }
            dialog.show()
            NetUtils.post(
                    BaseNetOption(COMMEND_ANSWER)
                            .addUserParam()
                            .param("answer_id", answer.id)
                            .param("content", commendStr),
                    object : NetCallback {
                        override fun onSuccess(data: JSONBean?) {
                            dialog.dismiss()
                            if ((data as BaseBean).isStatusOK()) {
                                et_commend.setText("")
                                et_commend.clearFocus()
                                refreshCommendData()
                            } else {
                                ToastUtils.show("评论失败")
                            }
                        }

                        override fun onFail(e: java.lang.Exception?) {
                            dialog.dismiss()
                            ToastUtils.show("网络错误")
                        }
                    }
            )
        }

        rv_question_detail.layoutManager = LinearLayoutManager(this)
        rv_question_detail.adapter = adapter
    }

    private fun refreshCommendData() {
        NetUtils.post(
                BaseNetOption(GET_COMMEND_LIST)
                    .addUserParam()
                    .param("answer_id", answer.id),
                object : NetCallback{
                    override fun onSuccess(data: JSONBean?) {
                        if ((data as BaseBean).isGoodJson()) {
                            refreshCommendView(data.data!!)
                        } else {
                            ToastUtils.show("服务器错误")
                            finish()
                        }
                    }
                    override fun onFail(e: java.lang.Exception?) {
                        ToastUtils.show("服务器错误")
                        finish()
                    }
                })
    }

    private fun refreshCommendView(data: String) {
        val commendList: ArrayList<QuestionDetail.AnswersBean.Commend> = ArrayList()
        commendList.add(QuestionDetail.AnswersBean.Commend())
        val commendArray = JSONArray(data)
        for (i: Int in 0 until commendArray.length()) {
            commendList.add(QuestionDetail.AnswersBean.
                    Commend(commendArray.getJSONObject(i)))
        }
        adapter.refresh(commendList)
    }

}